### PR TITLE
rancher on rke2 with terraform kvm script

### PR DESCRIPTION
**Pull request by skanakal**

### CODE REVIEW

File: rke2-rancher-kvm-tf/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+2. Terraform: Install Terraform on your local machine. Visit the official Terraform website for installation instructions.
+3. [Ubuntu Server Cloud Image](https://cloud-images.ubuntu.com/)
+
+# Usage
+To use this Terraform script, follow these steps:
+
+1. Clone this repository to your local machine:
+
+~~~
+git clone https://github.com/your-username/rke2-rancher-kvm-tf.git
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/README.md:18]: I think the link you want to put here is different, `rke2-rancher-kvm-tf` is not the name of this repo

**Review by vatsalparekh — CHANGES_REQUESTED** [review]

File: rke2-rancher-kvm-tf/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,12 @@
+rke2_os_image_path = "./ubuntu-18.04-server-cloudimg-amd64.img"
+rke2_cluster_storage_pool_path = "/tmp/rancher_rke2_tf"
+
+rke2_join_token = "secrettoken"
+
+rke2_server_ips = ["192.168.100.151"]
+rke2_server_vcpu = 8
+rke2_server_memory = 8096
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/terraform.tfvars:8]: Do we need these high defaults?

File: rke2-rancher-kvm-tf/variable.tf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+variable "rke2_os_image_path" {
+  description = "Path to the operating system image"
+  #default  = "./ubuntu-16.04-server-cloudimg-amd64-disk1.img"
+  default = "http://cloud-images.ubuntu.com/releases/23.04/release-20230420/ubuntu-23.04-server-cloudimg-amd64.img"
+
+}
+
+variable "rke2_cluster_storage_pool_path" {
+  description = "The directory where the pool will keep all its volumes"
+  default = "/home/skanakal/Downloads/rke2_cluster_storage"
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/variable.tf:39]: Can this be made to relative path of the user and not hardcoded username?

File: rke2-rancher-kvm-tf/variable.tf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+variable "install_rke2_version" {
+  description = "Select the rke2 version"
+  #https://github.com/rancher/rke2/tags
+  default = "v1.24.14+rke2r1"
+}
+
+variable "rancher_version" {
+  description = "select the rancher version"
+  default = "2.7.3"
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/variable.tf:55]: rancher_version in `rke2-rancher-kvm-tf/terraform.tfvars` file seems different, does this conflict anywhere?

File: rke2-rancher-kvm-tf/variable.tf [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  description = "The name of the rke2 masters"
+  default = "master"
+}
+
+######### Default Configuration for RKE2 cluster nodes ############
+
+variable "rke2_os_image_path" {
+  description = "Path to the operating system image"
+  #default  = "./ubuntu-16.04-server-cloudimg-amd64-disk1.img"
+  default = "http://cloud-images.ubuntu.com/releases/23.04/release-20230420/ubuntu-23.04-server-cloudimg-amd64.img"
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/variable.tf:33]: In `rke2-rancher-kvm-tf/terraform.tfvars` file, ubuntu 18.08 is used, all this cloud images version could point to one version variable

File: rke2-rancher-kvm-tf/rancher_provisioner.sh [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+log "Installing Cert Manager crds"
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.11.1/cert-manager.crds.yaml
+
+# Add Helm repositories for cert-manager
+log "Adding Helm repositories for cert-manager"
+helm repo add jetstack https://charts.jetstack.io
+helm repo update
+
+log "Installing Cert Manager"
+helm install cert-manager jetstack/cert-manager   --namespace cert-manager   --create-namespace   --version v1.11.1
```
Comment (vatsalparekh) [inline rke2-rancher-kvm-tf/rancher_provisioner.sh:77]: will we have to update the cert-manager version here with increase of Rancher or RKE version? regardless, can this be made into a var?

**Review by vatsalparekh — CHANGES_REQUESTED** [review]

File: rke2-rancher-kvm-tf/variable.tf (hunk 4 above)
Comment (skanakal) [reply rke2-rancher-kvm-tf/variable.tf:55]: If you have two different values for the same variable in both variables.tf and terraform.tfvars, Terraform will use the value defined in terraform.tfvars during the execution. The reason for this is that terraform.tfvars is designed to override the default values specified in variables.tf.

Here's the order in which Terraform processes variable values:

1. Default Value (if specified in variables.tf)
2. Value from terraform.tfvars (if provided)
3. Value from environment variables with the naming convention TF_VAR_<variable_name> (e.g., TF_VAR_my_variable)
4. Value from CLI variable input (e.g., using -var option)
5. Value from interactive prompts (if none of the above is provided)

So, if you have defined a variable in both files, Terraform will prioritize the value from terraform.tfvars over the value from variables.tf.

For more info read thru https://developer.hashicorp.com/terraform/language/values/variables#variable-definition-precedence

**Pull request closed without merging**